msteams-platform/concepts/build-and-test/deep-link-application.md

```diff
@@ -49,9 +74,81 @@ microsoftTeams.executeDeepLink("https://teams.microsoft.com/l/app/f46ad259-0fe5-
+
+App users can browse through content in Teams from your tab using TeamsJS library. This is useful if your tab needs to connect users with other content in Teams, such as to a channel, message, another tab, or to open a scheduling dialog. In few instances navigation can also be accomplished using the TeamsJS library and where possible use of the typed capabilities of the TeamsJS library is recommended.
+
+One of the benefits of using TeamsJS, particularly for Teams app that might run in other hosts (Outlook and Office), is that it's possible to check if the host supports the capability you're attempting to use. To check a host's support of a capability, you can use the `isSupported()` function associated with the namespace of the API. The TeamsJS library organizes APIs into capabilities by way of namespaces. For example, prior to usage of an API in the `pages` namespace, you can check the returned Boolean value from `pages.isSupported()` and take the appropriate action within the context of your app and apps UI.  
+
+For more information about capabilities and the APIs in the TeamsJS library, see [Building tabs and other hosted experiences with the TeamsJS library](~/tabs/how-to/using-teams-client-sdk.md#apis-organized-into-capabilities).
+
+You can configure deep links to browse within your app in one of the following two ways:
 
-It's possible to navigate within an app using TeamsJS. The following code demonstrates how to navigate to a specific entity within your Teams app.
+* [Configure deep link to browse within your app manually](#configure-deep-link-to-browse-within-your-app-manually)
+* [Configure deep link to a tab using TeamsJS library](#configure-deep-link-to-a-tab-using-teamsjs-library)
+
+#### Configure deep link to browse within your app manually 
+
+Use the following format for a deep link in a bot, connector, or message extension card:
+
+`https://teams.microsoft.com/l/entity/<appId>/<entityId>?tenantId=<tenantId>&webUrl=<entityWebUrl>&label=<entityLabel>&context=<context>`
+
+* If the bot sends a message containing a `TextBlock` with a deep link, then a new browser tab is opened when the user selects the link. This happens in Chrome and in the Teams desktop app, when they are running on Linux.
+
+* If the bot sends the same deep link URL into an `Action.OpenUrl`, then the Teams tab opens in the current browser tab when the user selects the link.
+
+The query parameters are:
+
+| Parameter name | Description | Example |
+|:------------|:--------------|:---------------------|
+| `appId`&emsp; | The ID from Teams Admin Center. |fe4a8eba-2a31-4737-8e33-e5fae6fee194|
+| `entityId`&emsp; | The ID for the item in the tab, which you provided when [configuring the tab](~/tabs/how-to/create-tab-pages/configuration-page.md). <br> * When generating a URL for deep linking, continue to use entity ID as a parameter name in the URL. <br> * When configuring the tab, the context object refers to the entityID as {page.id}. |Tasklist123|
```

Comment:
The description here is incorrect. The entityId is an ID for the tab itself. The subEntityId is the ID of items within the tab.